content/post/055-js-libp2p-0.26.md

```diff
@@ -0,0 +1,128 @@
+---
+date: 2019-08-07
+url: 2019-08-07-js-libp2p-0-26
+title: js-libp2p 0.26.0 released
+author: Jacob Heun
+---
+
+> Gossipsub support and Promisify API
+
+# 🔦 Highlights
+
+## 🗣 Gossipsub
+Thanks to the awesome work of the [ChainSafe](https://github.com/ChainSafe) team, [Gossipsub](https://github.com/ChainSafe/gossipsub-js) is here! If you are using Pubsub you can now switch to using Gossipsub instead of Floodsub. Have old peers you need to flood stuff to? Don't worry, Gossipsub will automatically fallback to Floodsub for peers that don't support it. See the API Changes section below for how to migrate your config over to using Gossipsub.
+
+## 🚉 Promisify
+As we migrate to async/await we are promisifying the Libp2p public methods. Several lower level libraries are currently leveraging libp2p for testing. Their transition to full async/await will be greatly helped by promisifying the libp2p api. Once the async/await changes are propagated up, we will remove Promisify in a future update, along with callback support, as libp2p will have full async/await support at that time. See the API Changes below to check out the Breaking Changes.
+
+# 🏗 API Changes
+
+## BREAKING CHANGES
+
+### Configuration
+
+#### Switch Options
+
+Configuration for `libp2p-switch` has changed. `blacklistTTL` and `blackListAttempts` have been changed to `denyTTL` and `denyAttempts` respectively. You can set these as follows:
+```js
+const libp2p = new Libp2p({
+  switch: {
+    denyTTL: 120e3,
+    denyAttempts: 5,
+  }
+  ...
+})
+```
+
+#### Pubsub
+
+**Subscribe**: To comply with the pubsub interface, the order of params for `libp2p.pubsub.subscribe` have been reordered. They were previously `topic, options, handler, callback`, and are now `topic, handler, options, callback`. If your implementation does not use `options` you should be able to ignore this change.
```

Comment:
"pubsub interface"? Does this mean the interface in `interface-ipfs-core` or something else?

Reply:
Yes, `interface-ipfs-core`, I will link to it.